src/engine/Sort.cpp

```diff
@@ -15,29 +15,36 @@ size_t Sort::getResultWidth() const { return _subtree->getResultWidth(); }
 // _____________________________________________________________________________
 Sort::Sort(QueryExecutionContext* qec,
            std::shared_ptr<QueryExecutionTree> subtree, size_t sortCol)
-    : Operation(qec), _subtree(subtree), _sortCol(sortCol) {}
+    : Operation(qec), _subtree(std::move(subtree)), _sortCol(sortCol) {}
```

Comment:
The previous code copies the `shared_ptr` twice: once since it is passed by value into the function and once
since the `_subtree` member is copy-initialized. The second copy is not needed and copying `shared_ptr` is more expensive than moving it, so the second version is to be preferred.

In general, this comes from a rule formulated by Scott  Meyer's, basically you
a) have a type that is always copied (we need to take a copy because we want to take shared ownership together with whoever passed it in).
b) and that is cheap to move (true for shared_ptr, since we do not have to change the atomic reference count).


In general this allows us to make our code efficient without having to create separate overloads for rvalues and lvalues. If we were for example to call this function with an rvalue reference shared ptr, aka
``` auto x = Sort(qec, std::move(some_shared_tree), 3) ```
we would get zero copies, which would not be possible if we for example made the function argument a `const &`.

Also, clang-tidy suggests this, since it knows about this rule. The functionality is exactly the same.